### ISSUE TITLE
Prompts >64 tokens produce corrupted output with the pinned FlashInfer 0.6.18 (SM120 sparse prefill); 10-line mask fixes it

### ISSUE THREAD

**Issue by caiovicentino** [0xSero/deepseek-v4.1-flash-4x-rtx-pro-6000#1]
## Summary

Thanks for the recipe — it boots and runs well on our 4x RTX PRO 6000 box. One thing the smoke tests do not catch: with the pinned image (FlashInfer 0.6.18) **every prompt longer than 64 tokens produced corrupted output**, while prompts of ≤ 62 tokens and long generations were perfect. Root cause is in FlashInfer's SM120 sparse-MLA prefill orchestrator (lanes past `topk_length` are not bounded; see flashinfer-ai/flashinfer#5075). I opened an upstream issue with the details: **flashinfer-ai/flashinfer#5095**.

Sharing here because (a) the repo's validation prompts are ~18 tokens, so this passes the smoke tests, and (b) a 10-line addition to your `flash_mla_sm120.py` fixes it and keeps the fast FlashInfer prefill.

## Symptom

`temperature=0` chat completions: 18–62 prompt tokens correct and deterministic; 65+ tokens garbage (English → Wikipedia-looking text, Portuguese → looping CJK fragments). Long decodes from short prompts are fine. Reproduces with EP1/EP4, `flashinfer_mxfp4`/`marlin`, DSpark on/off, `OFFLOAD_MODE=ram`/`nvme`. `SGLANG_SM120_FLASHMLA_BACKEND=triton` is correct everywhere (but prefill drops to ~2.5k tok/s).

## Fix that worked for us

In `runtime/flash_mla_sm120.py`, `_flash_mla_sm120_prefill`, right after `idx` / `extra_idx` are computed and before the extra-cache page split:

```python
if topk_length is not None:
    ar = torch.arange(idx.shape[-1], device=idx.device)
    idx = torch.where(ar[None, :] < topk_length.view(-1, 1), idx, torch.full_like(idx, -1))
if extra_idx is not None and extra_topk_length is not None:
    ar = torch.arange(extra_idx.shape[-1], device=extra_idx.device)
    extra_idx = torch.where(ar[None, :] < extra_topk_length.view(-1, 1), extra_idx, torch.full_like(extra_idx, -1))
```

After this: 16-point bisection (18→72 tokens), 29k-token prompts and 5.8k-token generations all correct; prefill ≈ 6.4k tok/s at 29k input; 16 concurrent streams ≈ 525 tok/s aggregate; batch-1 decode ≈ 100 tok/s (≈ 140 on code, where DSpark acceptance is ~4.2/5).

## Suggestion

Add a >64-token prompt (and ideally a few hundred tokens) to the validation set so this class of bug is caught, since the kernel boundary is exactly at 64 query tokens. Happy to send a PR with the mask plus the bisection script if you want it.
